### PR TITLE
fix(vats)!: migrate to use `@agoric/far`, remove deprecated REPL code

### PR DESCRIPTION
**Pull request by michaelfig**
refs: #4144

## Description

Show how to use `@agoric/far` in client code, namely the `packages/vats` package.

While we're there, remove the long-deprecated `ui-agent` code from the REPL, and endow exports from `@agoric/far`.

### Documentation Considerations

The `ui-agent` was an undocumented feature, and not used except for an ancient repl demo (since rewritten) a long time ago.

### CODE REVIEW

File: packages/vats/src/virtual-purse.js [hunk 1]
```diff
@@ -9,9 +8,9 @@ import '@agoric/notifier/exported.js';
 
 /**
  * @template T
- * @typedef {import('@agoric/eventual-send').EOnly<T>} EOnly An object roughly
- * of type T that is only allowed to be consumed via eventual-send.  This allows
- * the object to return Promises wherever it wants to, even when T demands a
+ * @typedef {import('@agoric/far').EOnly<T>} EOnly An object roughly of type T
+ * that is only allowed to be consumed via eventual-send.  This allows the
+ * object to return Promises wherever it wants to, even when T demands a
```
Comment (dckc) [inline packages/vats/src/virtual-purse.js]: any particular reason to duplicate this documentation here?

**Review by dckc — COMMENTED** [review]
looks good so far, but the ci bots show this needs more work

File: packages/vats/src/virtual-purse.js (hunk 1 above)
Comment (michaelfig) [reply packages/vats/src/virtual-purse.js]: Not really.  I've removed this comment in favour of just a bare `@typedef`.

**Comment by michaelfig** [comment]
To do this migration, replace all possible instances of `@agoric/eventual-send` and `@agoric/marshal` with `@agoric/far` in vat code and libraries.

If `@agoric/far` does not export everything your module needs, it may require direct access to the original modules, but that is likely only necessary for infrastructure like `SwingSet`, `install-ses` and `captp`.

**Review by dckc — APPROVED** [review]
yay! more deletes than adds!

including a whole file of technical debt paid off

**Pull request merged**